src/app/dashboard/constraint-builder-overlay/constraint-builder.component.ts

```diff
@@ -0,0 +1,163 @@
+import {
+  ConstraintFunction,
+  PropertySelectGroup,
+  SelectData,
+} from 'src/app/shared/matching/constraints/constraint-functions/constraint-function';
+import { StudentsService } from 'src/app/shared/data/students.service';
+import { ConstraintBuilderService } from 'src/app/shared/matching/constraints/constraint-builder';
+import { SkillsService } from 'src/app/shared/data/skills.service';
+import { ProjectsService } from 'src/app/shared/data/projects.service';
+import { ConstraintWrapper } from 'src/app/shared/matching/constraints/constraint';
+import { ConstraintsService } from 'src/app/shared/data/constraints.service';
+import { Project, Skill, Student } from 'src/app/api/models';
+import { Subscription } from 'rxjs';
+import { Operator } from 'src/app/shared/matching/constraints/constraint-utils';
+
+@Component({
+  selector: 'app-constraint-builder',
+  templateUrl: './constraint-builder.component.html',
+  styleUrl: './constraint-builder.component.scss',
+})
+export class ConstraintBuilderComponent implements OverlayComponent, OnInit, OnDestroy {
+  // change OverlayComponent in future
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  data: any;
+  constraintFunctionProperties: PropertySelectGroup[];
+  constraintFunctionOperatorSelectData: SelectData[];
+  constraintFunctionValueSelectData: SelectData[];
+  constraintProjectSelectData: SelectData[];
+  constraintOperators: SelectData[];
+
+  selectedConstraintFunctionProperty: string;
+  selectedConstraintFunctionOperator: string;
+  selectedConstraintFunctionValue: string;
+  selectedConstraintProject: string;
+  selectedConstraintOperator: string;
+  constraintThresholdValue: number;
+
+  constraint?: ConstraintWrapper;
+
+  private students: Student[];
+  private skills: Skill[];
+  private projects: Project[];
+  private subscriptions: Subscription[] = [];
+
+  private constraintFunctions: ConstraintFunction[];
+  private readonly CONSTRAINT_OPERATORS = [Operator.GREATER_THAN_OR_EQUAL, Operator.LESS_THAN_OR_EQUAL];
+
+  constructor(
+    private constraintsService: ConstraintsService,
+    private overlayService: OverlayService,
+    private constraintBuilderService: ConstraintBuilderService,
+    private studentsService: StudentsService,
+    private skillsService: SkillsService,
+    private projectsService: ProjectsService
+  ) {}
+
+  ngOnInit(): void {
+    this.subscriptions.push(
+      this.studentsService.students$.subscribe(students => {
+        this.students = students;
+        this.updateData();
+      }),
+      this.skillsService.skills$.subscribe(skills => {
+        this.skills = skills;
+        this.updateData();
+      }),
+      this.projectsService.projects$.subscribe(projects => {
+        this.projects = projects;
+        this.updateData();
+      })
+    );
+  }
+
+  ngOnDestroy(): void {
+    this.subscriptions.forEach(subscription => subscription.unsubscribe());
```

Comment:
subscription could be undefined. Simply add a `?`. --> `subscription?.unsubscribe`